### DISCUSSION TITLE
tag_queries overwrite

### DISCUSSION THREAD

**Discussion by spham** [rabbitmq/rabbitmq-server#4123] [Questions]
Hi
when I use $vhost in tag_queries, the behavior is strange.
```
[
    {rabbitmq_auth_backend_ldap, [
        {resource_access_query, {in_group, "CN=${vhost},OU=groups,DC=bakken,DC=io"}},
        {vhost_access_query, {in_group, "CN=${vhost},OU=groups,DC=bakken,DC=io"}},
        {tag_queries, [
            {administrator, {in_group, "CN=${vhost},OU=groups,DC=bakken,DC=io"}},
            {management, {constant, true}}
        ]}
    ]}
].
```

`2022-02-09 12:43:50.787309+00:00 [info] <0.392.0>     LDAP evaluated in_group ...  true`

that mean all seems ok, but curl return

`{"name":"...","tags":["management"]}`
in this case, ui management, give access to some vhost but as management role, only :-(

it's like foreach administrator, give management role
`"administrator: true, management: true}" then management`

![image](https://user-images.githubusercontent.com/1083631/153207594-26f5920e-4ac0-42df-9ee0-fbf4ff60b09d.png)

in other case

```
[
    {rabbitmq_auth_backend_ldap, [
        {resource_access_query, {in_group, "CN=${vhost},OU=groups,DC=bakken,DC=io"}},
        {vhost_access_query, {in_group, "CN=${vhost},OU=groups,DC=bakken,DC=io"}},
        {tag_queries, [
            {administrator, {constant, true}},
            {management, {constant, true}}
        ]}
    ]}
].

```
curl return
`{"name":"...","tags":["administrator","management"]}`
in this case, ui management, have differend behavior
user can full access to all vhost as administrator, but not limit by vhost

![image](https://user-images.githubusercontent.com/1083631/153207856-4771b900-6018-425a-915a-b642c8e611a9.png)

- Rabbitmq 4.9.13
- Erlang 23.3.4
- Centos 7
- OpenJDK 11

**Comment by lukebakken** [comment]
I have already said that not all operations have a `vhost` associated with them - https://github.com/rabbitmq/rabbitmq-server/discussions/4021#discussioncomment-2039246

https://groups.google.com/g/rabbitmq-users/c/jhvt37Byixo/m/gL_3A3onBAAJ

```
curl -i -u http://localhost:15672/api/whoami
```

The `curl` API request does not have a `vhost` associated with it, thus there is no `vhost` value to put into the `${vhost}` template in your LDAP query, thus there never will be an `administrator` tag returned.

Please STOP posting in multiple forums. I will _only_ follow up in this discussion from here on out:

https://github.com/rabbitmq/rabbitmq-server/discussions/4021
